contrib/ansible/Getting-Started.md

```diff
@@ -0,0 +1,156 @@
+
+
+# Kubernetes Cluster with Containerd and Cri-Containerd
+
+This document provides the steps to bring up a Kubernetes cluster using ansible and kubeadmusing ansible and kubeadm tools. 
+
+### Prerequisites:
+- **OS**: Ubuntu 16.04 (will be updated with additional distros after testing)
+- **Python**: 2.7+
+- **Ansible**: 2.3+
+
+## Step 0:
+-  Install Ansible on any one of the nodes in the candidate cluster or an external machine which will be used to provision the cluster. The install instructions can be found [here](http://docs.ansible.com/ansible/latest/intro_installation.html). The playbooks in the repo need ansible version 2.3 or later.
+-  Create a hosts file and include the Routable IPs of the nodes that need to be provisioned by ansible.
+```
+$ cat hosts
+172.31.7.230
+172.31.13.159
+172.31.1.227
+```
+-  Setup passwordless SSH access from ansible control machine to all the hosts in the hosts file. The instructions can be found in [here](http://www.linuxproblem.org/art_9.html)
+
+## Step 1:
+At this point the ansible playbook should be able to ssh into all the machines in the hosts file and start installing the playbook
+
+```
+git clone https://github.com/kubernetes-incubator/cri-containerd
+cd ./cri-containerd/contib/ansible
+ansible-playbook -i hosts cri-containerd.yaml
+```
+A typical cloud login might have a user and private key file in which case the following can be used
+```
+ansible-playbook -i hosts -u <username> --private-key <example.pem file> cri-containerd.yaml
+ ``` 
+For more options ansible config file (/etc/ansible/ansible.cfg) can be use to set defaults.Please refer to ansible [options](http://docs.ansible.com/ansible/latest/intro_configuration.html) for advanced ansible configurations.
+
+At the end of this step you will have all the hosts installed with the necessary components to bringup a kubernetes cluster
+with containerd and cri-containerd. Cri-Containerd, Containerd, kubelet, kubeadm, kubectl and other dependencies will be installed and configured.
+
+```
+PLAY RECAP ***************************************************************************************************************************************************************
+172.31.1.227               : ok=21   changed=7    unreachable=0    failed=0
+172.31.13.159              : ok=21   changed=7    unreachable=0    failed=0
+172.31.7.230               : ok=21   changed=8    unreachable=0    failed=0
+```
+
+## Step 2:
+Use [kubeadm](https://kubernetes.io/docs/setup/independent/install-kubeadm/) tool from kubernetes to intiate and form a kubernetes cluster.
+Initialize the cluster from one of the nodes.
+```
+$sudo kubeadm init --skip-preflight-checks
```

Comment:
We may want to point out that for some network provider `--pod-network-cidr` is needed here, e.g. flannel.

kubeadm doc phrases it like this.
`Depending on what third-party provider you choose, you might have to set the --pod-network-cidr to something provider-specific`